tariff/octopus.go

```diff
@@ -62,11 +80,35 @@ func (t *Octopus) run(done chan error) {
 	client := request.NewHelper(t.log)
 	bo := newBackoff()
 
+	var restQueryUri string
+
+	// If ApiKey is available, use GraphQL to get appropriate tariff code before entering execution loop.
+	if t.apikey != "" {
```

Comment:
Shouldn't this part be moved to the tariff init as its a one-time setup efforts? Also saves storing the api key and client.

Reply:
On mobile but I'll inspect this in more detail once I'm in front of a laptop later 🙏

Reply:
ping ;)

Reply:
Sorry for the delay, life's been happening 😭

You've a fair point here - was trying to think how best to handle these two ways of setting up. Refactoring the `rest` module would probably be a better solution, generating the API URL and holding onto it in this execution loop seems a bit wrong.

The above being said, see my other comment on refresh tokens.